fuel-chain-config/src/config/chain.rs

```diff
@@ -123,6 +129,31 @@ impl FromStr for ChainConfig {
     }
 }
 
+impl GenesisCommitment for ChainConfig {
+    fn root(&mut self) -> anyhow::Result<MerkleRoot> {
+        // TODO: Hash settlement configuration, consensus block production
+        let config_hash = *Hasher::default()
+            .chain(self.block_gas_limit.to_be_bytes())
+            .chain(self.transaction_parameters.root()?)
+            .chain(self.chain_name.as_bytes())
+            .chain(bincode::serialize(&self.initial_state)?)
```

Comment:
The root/hash of the initial state is included in the `Genesis` struct, so we don’t need it here:)

Reply:
I see, I misunderstood @Voxelot's [comment](https://github.com/FuelLabs/fuel-core/issues/824#issuecomment-1344801975) - you mean we will add it as part of the Gensis in another PR?

Reply:
I meant `Genesis` already has `chain_config_hash` and roots of all states, the hash of `Genesis` covers all our needs:) We can use it as unique chain identifier, aka checksum